File: app/src/main/java/top/mikoto/sangnam/Utils/PrefManager.java
package top.mikoto.sangnam.Utils;

import android.content.Context;
import android.content.SharedPreferences;

public class PrefManager {
    private final SharedPreferences pref;
    private final SharedPreferences.Editor editor;
    private final Context _context;

    // shared pref mode
    private final int PRIVATE_MODE = 0;

    // Shared preferences file name
    private static final String PREF_NAME = "sangnam";

    private static final String IS_FIRST_TIME_LAUNCH = "IsFirstTimeLaunch";
    private static final String LATEST_ID = "LatestID";

    public PrefManager(Context context) {
        this._context = context;
        pref = _context.getSharedPreferences(PREF_NAME, PRIVATE_MODE);
        editor = pref.edit();
    }

    public void setFirstTimeLaunch(boolean isFirstTime) {
        editor.putBoolean(IS_FIRST_TIME_LAUNCH, isFirstTime);
        editor.commit();
    }

    public boolean isFirstTimeLaunch() {
        return pref.getBoolean(IS_FIRST_TIME_LAUNCH, true);
    }


}

File: app/src/main/java/top/mikoto/sangnam/utils/PrefManager.kt
package top.mikoto.sangnam.utils

import android.content.Context
import android.content.SharedPreferences
import android.content.SharedPreferences.Editor

class PrefManager(_context: Context) {
    private val pref: SharedPreferences
    private val editor: Editor

    companion object {
        // Shared preferences file name
        private const val PREF_NAME = "sangnam"
        private const val IS_FIRST_TIME_LAUNCH = "IsFirstTimeLaunch"
        private const val LATEST_ID = "LatestID"
    }

    init {
        pref = _context.getSharedPreferences(PREF_NAME, 0)
        editor = pref.edit()
    }
}